File: main/bon-appetit/bon-appetit.cs
using System;
using System.Linq;

internal static class Solution {
    private static int[] GetRecord()
            => Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);

    private static void Main()
    {
        var nk = GetRecord();

        var k = nk[1];
        var c = GetRecord();
        var b_charged = int.Parse(Console.ReadLine());

        var b_actual = (c.Sum() - c[k]) / 2;

        if (b_charged > b_actual)   Console.WriteLine(b_charged - b_actual);
        else                        Console.WriteLine("Bon Appetit");
    }
}

File: main/bon-appetit/bon-appetit.pl
#!/usr/bin/env perl

use strict;
use warnings;
use Carp qw(croak);
use List::Util qw(sum);

sub read_record {
    return split / /ms, <>;
}

my ($n, $k) = read_record;
my @c = read_record;
scalar @c == $n or croak 'wrong record length';
my $b_charged = <>;

my $b_actual = (sum(@c) - $c[$k]) / 2;

if ($b_charged == $b_actual) {
    print "Bon Appetit\n";
} else {
    printf "%d\n", $b_charged - $b_actual;
}
